src/algorithms/all_shortest_paths.c

```diff
@@ -27,21 +27,27 @@ int AllShortestPaths_FindMinimumLength
 	NodeID destID = ENTITY_GET_ID(dest);
 
 	GrB_Vector visited;
+	GrB_Vector current_visited;
 
 	GrB_Vector_new(&visited, GrB_BOOL, Graph_UncompactedNodeCount(ctx->g));
 	GxB_set(visited, GxB_SPARSITY_CONTROL, GxB_BITMAP);
 
+	GrB_Vector_new(&current_visited, GrB_BOOL, Graph_UncompactedNodeCount(ctx->g));
+	GxB_set(current_visited, GxB_SPARSITY_CONTROL, GxB_BITMAP);
+
 	while (true) {
 		// see if we have nodes in the current level
 		uint neighborCount = array_len(ctx->levels[depth]);
 		if(neighborCount == 0) {
 			// move to next level
 			depth++;
-			if(depth > ctx->maxLen || depth >= array_len(ctx->levels)) {
+			if(depth == ctx->maxLen || depth >= array_len(ctx->levels)) {
```

Comment:
Is the change to `==` valid? `depth` was just incremented, I would expect that we still need to check the current element.

Reply:
yes this is bfs so we not going to increment more then what needed